platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/properties/PropertiesLoader.java

```diff
@@ -108,108 +125,133 @@ public Properties loadProperties(String propertiesFile) {
+    do {
+      properties = strategiesIterator.next().apply(propertiesFile, classLoader);
+    } while (properties.isEmpty() && strategiesIterator.hasNext());
 
-      // replace any ${prop} with system properties
-      Properties filtered = new Properties();
-      for (Map.Entry<?, ?> entry : properties.entrySet()) {
-        filtered.put(
-            StrSubstitutor.replaceSystemProperties(entry.getKey()),
-            StrSubstitutor.replaceSystemProperties(entry.getValue()));
+    properties = substituteSystemPropertyPlaceholders(properties);
+    return properties;
+  }
+
+  /** Default property loading strategy. */
+  @SuppressWarnings("squid:S1172" /* Used in bi-function */)
+  @VisibleForTesting
+  static Properties attemptLoadWithSpring(String propertiesFile, ClassLoader classLoader) {
+    Properties properties = new Properties();
+    try {
+      LOGGER.debug(
+          "Attempting to load properties from {} with Spring PropertiesLoaderUtils.",
+          propertiesFile);
+      properties = PropertiesLoaderUtils.loadAllProperties(propertiesFile);
+    } catch (IOException e) {
+      LOGGER.debug("Unable to load properties using default Spring properties loader.", e);
+    }
+    return properties;
+  }
+
+  /** Try loading properties using Spring and a provided class loader. */
+  @VisibleForTesting
+  static Properties attemptLoadWithSpringAndClassLoader(
+      String propertiesFile, ClassLoader classLoader) {
+    Properties properties = new Properties();
+    try {
+      LOGGER.debug(
+          "Attempting to load properties from {} with Spring PropertiesLoaderUtils with class loader.",
+          propertiesFile);
+      if (classLoader != null) {
+        properties = PropertiesLoaderUtils.loadAllProperties(propertiesFile, classLoader);
+      } else {
+        properties =
+            PropertiesLoaderUtils.loadAllProperties(
+                propertiesFile, PropertiesLoader.class.getClassLoader());
       }
-      properties = filtered;
+    } catch (IOException e) {
+      LOGGER.debug("Unable to load properties using default Spring properties loader.", e);
+    }
+    return properties;
+  }
+
+  /**
+   * Try loading the properties directly from the file system. If the properties file has a
+   * fully-qualified absolute path (which is what the blueprint file should specify) then it can be
+   * loaded directly, using this method. Otherwise the path will be considered relative and attempts
+   * will be made with {@code karaf.home} and {@code ddf.home} property values prepended to the
+   * original path.
+   */
+  @SuppressWarnings("squid:S1172" /* Used in bi-function */)
+  @VisibleForTesting
+  static Properties attemptLoadWithFileSystem(String propertiesFile, ClassLoader classLoader) {
+    LOGGER.debug("Attempting to load properties from file system: {}", propertiesFile);
+    Properties properties = new Properties();
+
+    List<String> systemProperties = ImmutableList.of("karaf.home", "ddf.home");
```

Comment:
❗️ Yeah I understand what you were going for but I can't justify these changes when the base list won't be flexing much, if **at all**, over the course of time.